src/test/scala/viper/gobra/typing/StmtTypingUnitTests.scala

```diff
@@ -0,0 +1,70 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) 2011-2020 ETH Zurich.
+
+package viper.gobra.typing
+
+import org.scalatest.Inside
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.funsuite.AnyFunSuite
+import viper.gobra.ast.frontend._
+import viper.gobra.frontend.info.implementation.TypeInfoImpl
+import org.bitbucket.inkytonik.kiama.util.Positions
+import viper.gobra.frontend.PackageInfo
+import viper.gobra.frontend.info.Info
+import viper.gobra.frontend.Config
+
+class StmtTypingUnitTests extends AnyFunSuite with Matchers with Inside {
+  val frontend = new TestFrontend()
+
+  test("TypeChecker: should detect labeled continues outside of loops") {
+    assert(!frontend.wellDefStmt(PContinue(Some(PLabelUse("l"))))().valid)
+  }
+
+  test("TypeChecker: should detect continues outside of loops") {
+    assert(!frontend.wellDefStmt(PContinue(None))().valid)
+  }
+
+  test("TypeChecker: should detect labeled breaks outside of loops") {
+    assert(!frontend.wellDefStmt(PBreak(Some(PLabelUse("l"))))().valid)
+  }
+
+  test("TypeChecker: should detect breaks outside of loops") {
+    assert(!frontend.wellDefStmt(PBreak(None))().valid)
+  }
+
+  class TestFrontend {
+    def singleStmtProgram(inArgs: Vector[(PParameter, Boolean)], body : PStatement) : PProgram = PProgram(
+      PPackageClause(PPkgDef("pkg")),
+      Vector(),
+      Vector(PMethodDecl(
+        PIdnDef("foo"),
+        PUnnamedReceiver(PMethodReceiveName(PNamedOperand(PIdnUse("self")))),
+        inArgs.map(_._1),
+        PResult(Vector()),
+        PFunctionSpec(Vector(), Vector(), Vector(), Vector(), isPure = false),
+        Some(PBodyParameterInfo(inArgs.collect{ case (n: PNamedParameter, true) => PIdnUse(n.id.name) }), PBlock(Vector(body)))
+      ))
```

Comment:
Please add the program as a comment so that it is easier to read, e.g. 
```
package pkg

func (whatever the receiver is) foo(<arguments>) {
  <body>
}
```

It seems to me that the receiver is incorrect. You should have positives tests (where you assert that typechecking is successful) as well, to make sure that the tests do not fail due to something else, e.g. an incorrect receiver.